20150310-wp-core-contrib-day.md

```diff
@@ -0,0 +1,255 @@
+
+![WordPress Core Site Setup (MAMP PRO)](https://s3.amazonaws.com/cdn.deliciousbrains.com/content/uploads/2015/03/WordPress-Core-Site-Setup-MAMP-PRO.png)
+
+### Give It a Spin
+The moment of truth, make sure you can connect to your new WordPress development site and set it up.
+
+![WordPress › Installation](https://s3.amazonaws.com/cdn.deliciousbrains.com/content/uploads/2015/03/WordPress-%E2%80%BA-Installation.png)
+
+![Dashboard ‹ WordPress Core — WordPress](https://s3.amazonaws.com/cdn.deliciousbrains.com/content/uploads/2015/03/Dashboard-%E2%80%B9-WordPress-Core-—-WordPress.png)
+
+![WordPress Core | Just another WordPress site](https://s3.amazonaws.com/cdn.deliciousbrains.com/content/uploads/2015/03/WordPress-Core-Just-another-WordPress-site.png)
+
+### Run The Unit Tests
+It's a very good idea to run the unit tests just to be sure all is well in your development site, and to make sure any changes you later make do not break anything.
+
+From the root of the checked out source...
+
+```
+phpunit
+```
+
+![WordPress Core Unit Tests Run](https://s3.amazonaws.com/cdn.deliciousbrains.com/content/uploads/2015/03/WordPress-Core-Unit-Tests-Run.png)
+
+Don't be surprised if you see a failure or two in the unit tests, during development there often are some, maybe you could fix them? At least you know where the problems exist and so when you make your changes you can tell whether you made things better, worse, or at least didn't affect them!
+
+## Development
+You're now all set up to develop on WordPress Core, keep your repository up to date with `svn up` and rebuild your WordPress Core site with `grunt` as required or through a `watch` process.
+
+Let's assume you're starting off your WordPress Core contributions by refreshing a patch, this entails downloading the patch file from a ticket, applying it to your checked out source, checking that it still applies correctly and does what is expected against the latest development version of WordPress Core, and then re-creating the patch and attaching to the ticket with an updated name and updated keywords.
+
+### Keep It Clean
+Before you apply a patch, be absolutely sure that your checked out source and build are "clean", there is nothing worse than working on something only to find that you can't create a clean patch because you have some previous code changes from another ticket polluting your code base.
+
+```
+svn diff
+```
+
+If `svn diff` returns anything, maybe because you've already worked on a patch, revert the changes.
+
+```
+svn revert path/to/changed/file
+```
+
+And then double check again with `svn diff`.
+
+When there are no changes in your checkout, make sure you're up to date with other changes that have been committed to the repository.
+
+```
+svn up
+```
+
+Now ensure your cleaned up repository builds, and then [Give It a Spin][].
+
+```
+grunt
+```
+
+### Applying a Patch
```

Comment:
If you don't mind I'll skip this tip as I've not used it and I'd like to keep the article as simple as possible.